nltk/translate/lepor.py

```diff
@@ -0,0 +1,253 @@
+    :type reference_length: int
+    :param hypothesis_length: length of the hypothesis sentence
+    :type hypothesis_length: int
+    :param alpha: a parameter to set weight fot recall.
+    :type alpha: float
+    :param beta: a parameter to set weight fot precision.
+    :type beta: float
+
+    :return: harmonic mean.
+    :rtype: float
+    """
+
+    epsilon = sys.float_info.epsilon
+
+    precision = match_count / hypothesis_length
+    recall = match_count / reference_length
+
+    harmonic_score = (alpha + beta) / (
+        (alpha / (recall + epsilon)) + (beta / (precision + epsilon))
+    )
+
+    return harmonic_score
+
+
+def sentence_lepor(
+    reference: str, hypothesis: str, alpha: float = 1.0, beta: float = 1.0
+) -> float:
+    """
+    Calculate LEPOR score a sentence from Han, A. L.-F. (2017).
+    LEPOR: An Augmented Machine Translation Evaluation Metric. https://arxiv.org/abs/1703.08748v2
+
+    >>> hypothesis = 'a bird is on a stone.'
+
+    >>> reference = 'a bird behind the stone.'
+
+    >>> sentence_lepor(reference, hypothesis)
+    0.7824248013113159
+
+    :param reference: reference sentence
+    :type reference: list(list(str))
+    :param hypothesis: a hypothesis sentence
+    :type hypothesis: list(str)
+    :param alpha: a parameter to set weight fot recall.
+    :type alpha: float
+    :param beta: a parameter to set weight fot precision.
+    :type beta: float
+
+    :return: The Lepor score. Returns a float value of lepor score for a sentence
+    :rtype: float
+
+    """
+
+    if len(reference) == 0 or len(hypothesis) == 0:
+        raise ValueError("One of the sentence is empty. Exit.")
+
+    # Regex to remove all the unnecessary punctuation from the text.
+    reference = nltk.word_tokenize(reference)
+    hypothesis = nltk.word_tokenize(hypothesis)
+
+    # reference = re.findall(r"[\w']+|[.,!?;]", reference)
+    # hypothesis = re.findall(r"[\w']+|[.,!?;]", hypothesis)
+
+    # Calculate the length penalty due to the difference in the length of reference and hypothesis.
+    lp = length_penalty(reference, hypothesis)
+
+    # Calculate the penalty on different positions of same word in translation.
+    npd, match_count = ngram_positional_penalty(reference, hypothesis)
+
+    harmonic_score = harmonic(match_count, len(reference), len(hypothesis), alpha, beta)
+
+    return lp * npd * harmonic_score
+
+
+def corpus_lepor(
+    references: List[str], hypothesis: List[str], alpha: float = 1.0, beta: float = 1.0
+) -> list[float]:
+    """
+    Calculate LEPOR score for list of sentences from Han, A. L.-F. (2017).
+    LEPOR: An Augmented Machine Translation Evaluation Metric. https://arxiv.org/abs/1703.08748v2
+
+    >>> hypothesis = ['a bird is on a stone.', 'scary crow was not bad.']
+
+    >>> references = ['a bird behind the stone.', 'scary cow was good.']
+
+    >>> corpus_lepor(references, hypothesis)
+    [0.7824248013113159, 0.5639427891892225]
+
+
+    :param references: reference sentences
```

Comment:
Same comment as the sentence_lepor, the expected behavior should take in multiple references per hypothesis